lti_consumer/lti_xblock.py

```diff
@@ -1687,10 +1700,8 @@ def _get_lti_1p3_launch_url(self, consumer):
         """
         lti_1p3_launch_url = self.lti_1p3_launch_url.strip()
 
-        # The "external" config_type is not supported for LTI 1.3, only LTI 1.1. Therefore, ensure that we define
-        # the lti_1p3_launch_url using the LTI 1.3 consumer only for config_types that support LTI 1.3.
-        # TODO: This needs to change when the LTI 1.3 support is added to the external config_type in the future.
-        if consumer and self.lti_version == "lti_1p3" and self.config_type == "database":
+        # Get LTI launch URL from consumer if using database or external configuration type.
+        if consumer and self.lti_version == 'lti_1p3' and self.config_type in ('database', 'external'):
```

Comment:
Something odd happens when I try to use the external configuration directly from the XBlock. When I set the `Configuration Type` to `Reusable Configuration` and provide a valid `LTI Reusable Configuration ID`, the XBlock break with `Error: (1048, "Column 'version' cannot be null")`. The relevant part of the traceback is:
```python
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/lti_xblock.py", line 1177, in author_view
  return self.student_view(context)
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/lti_xblock.py", line 1227, in student_view
  context.update(self._get_context_for_template())
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/lti_xblock.py", line 1730, in _get_context_for_template
  lti_consumer = self._get_lti_consumer()
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/lti_xblock.py", line 1128, in _get_lti_consumer
  return get_lti_consumer(config_id_for_block(self))
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/api.py", line 96, in config_id_for_block
  config = _get_lti_config_for_block(block)
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/api.py", line 76, in _get_lti_config_for_block
  lti_config = _get_or_create_local_lti_config(
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/api.py", line 46, in _get_or_create_local_lti_config
  lti_config.save()
File "/edx/shared-src/xblock-lti-consumer/lti_consumer/models.py", line 313, in save
  super().save(*args, **kwargs)
```

Reply:
This is because the external ID set on the XBlock is not found, this is either because the reusable configuration feature is not properly setup or the external configuration doesn't exist. If you check this [specific line](https://github.com/openedx/xblock-lti-consumer/blob/d315f5afb6f01c0efc4534dab7ba88bcf9bb0ce2/lti_consumer/api.py#L77) you can notice that if no config is found the "version" value will return None, once this value is sent to the `_get_or_create_local_lti_config` function, when it tries to save the new values to the `LtiConfiguration` it fails since "version" value cannot be null. This could be fixed by adding an extra validation to any required field from the external config on `_get_lti_config_for_block` but I think it's out of the scope of this PR, an issue about this should be created.

Reply:
Created #420.